### PR TITLE
Adding Scoop support

### PR DESCRIPTION
**Pull request by CEbbinghaus**
This PR is to add scoop support to the CLI install.
With first a github hosted uri to install it manually:
```
> scoop install https://raw.githubusercontent.com/appwrite/sdk-for-cli/master/appwrite.json
```
and hopefully the ability for scoop to verify hashes as well as install it via the [main bucket](https://github.com/ScoopInstaller/Main)
```
> scoop install appwrite
```
which will require a working autoupdater and working version check

This PR implements Feature appwrite/sdk-for-cli/issues/39

### CODE REVIEW

File: src/SDK/Language/CLI.php
Comment (Meldiron) [inline src/SDK/Language/CLI.php:79]: Can we name it something else? We use the file `appwrite.json` for project configuration used by Appwrite commands (`init, deploy`). It cold create some confusions when looking at CLI SDK

Comment (Meldiron) [reply src/SDK/Language/CLI.php:79]: My idea: `scoop.json`

**Review by Meldiron — COMMENTED** [review]
Left a comment, PR looks good! 🥳 (didn't test)

File: templates/cli/appwrite.json.twig
Comment (stnguyen90) [inline templates/cli/appwrite.json.twig:14]: I don't think this exists. Where's the update to the job that will generate this.

File: templates/cli/appwrite.json.twig
Comment (stnguyen90) [inline templates/cli/appwrite.json.twig:21]: I don't think this exists. Where's the update to the job that will generate this.

File: templates/cli/appwrite.json.twig
Comment (stnguyen90) [inline templates/cli/appwrite.json.twig:6]: Scoop let's you specify architecture as well. It would probably be good to differentiate between arm and x86.

**Review by stnguyen90 — CHANGES_REQUESTED** [review]

File: src/SDK/Language/CLI.php
Comment (CEbbinghaus) [reply src/SDK/Language/CLI.php:79]: I thought the same thing but since the manifest doesn't contain a name the name of the file is the name that it installs the program under. I tried using scoop.json but when I tried it actually uninstalled scoop because it thought this was a new version of it. So to have the program called appwrite it has to be appwrite.json. A little annoying but I think if we keep it in a folder called scoop it could work

File: templates/cli/appwrite.json.twig
Comment (CEbbinghaus) [reply templates/cli/appwrite.json.twig:6]: Nope. The architecture difference is x32 vs x64. This is addressed in issue ScoopInstaller/Scoop#3146, however, that is not fixed yet. I am subscribed to it though so as soon as they add support for ARM I will update the manifest to reflect those changes.

File: templates/cli/appwrite.json.twig
Comment (CEbbinghaus) [reply templates/cli/appwrite.json.twig:14]: You are right. This doesn't exist right now and I have no idea why it's here since I had taken it out to commit the file. My bad that it got pushed but it will be removed until a proper system has been figured out.

File: src/SDK/Language/CLI.php
Comment (stnguyen90) [reply src/SDK/Language/CLI.php:79]: Scoop folder would be good.

File: templates/cli/appwrite.json.twig
Comment (CEbbinghaus) [reply templates/cli/appwrite.json.twig:21]: Wasn't meant to be committed. Will be removed until that update has been figured out

File: templates/cli/appwrite.json.twig
Comment (CEbbinghaus) [reply templates/cli/appwrite.json.twig:14]: yes, however, if we want those changes reflected in the JSON file in the commit then the workflow will have to create another commit on the master branch updating the hash. (which personally I don't find very appealing). The two other solutions would be providing the JSON as a release artifact (which I would think is a bit strange) or leaving away the hash for the base file and only having the auto-updater.

The package will work without the hash it just misses the client-side verification but that is only relevant when expecting an MTM attack which is going to be unlikely. Instead, I propose we create the file with the hash embedded and create an automated PR to the main bucket of the scoop installer to update the app there. We don't even have to do this all the time since the auto updater can get the hashes from the URL.

**Comment by CEbbinghaus** [comment]
Install can be tested with this command:
```powershell
$ scoop install https://gist.githubusercontent.com/CEbbinghaus/3219a010325e9684839f7ce5ecea4afd/raw/2f19d33dc4581658d5e73edda4d763644129621a/appwrite.json
```

Generated with the SDK

**Comment by CEbbinghaus** [comment]
The build task seems to be failing because of Docker rate limits not the changes I added

**Review by Meldiron — APPROVED** [review]
Looks good to me 😇

**Comment by Meldiron** [comment]
> The build task seems to be failing because of Docker rate limits not the changes I added

Restarted failed jobs. All green now 💚

**Comment by CEbbinghaus** [comment]
Sweet. This PR is ready to merge then as it shouldn't contain any extra changes. It's going to be the prerequisite for the other changes so i am very keen to get it merged asap

**Comment by CEbbinghaus** [comment]
Requesting another review as its still not merged and the merging is blocked. It might be smart to leave it until ScoopInstaller/Scoop#3146 is merged as it would allow for the ARM release to be installed too

**Comment by lohanidamodar** [comment]
@Meldiron looks like the blocking issue was merged can you help move forward with this?

**Comment by CEbbinghaus** [comment]
Thanks @lohanidamodar, since opening this PR I have learned a lot about scoop & how it works. There are some small modifications I want to do to it as it stands right now since it would require a PullRequest to the scroop repository for every version which is not desired. Instead if we can generate the Appwrite.json file a little differently, it should auto update within the scoop repository. That means that the changes will be lagging behind until scoop's main bucket does another update cycle but that is probably faster than waiting for a human review of every version.

File: templates/cli/scoop/appwrite.json.twig
Comment (CEbbinghaus) [inline templates/cli/scoop/appwrite.json.twig:17]: Since the sdk-for-cli repository now does its releases through actual releases rather than Pre-Releases this can be simplified to just point at the git repository. this removes the burden of maintaining this configuration & keeping it up to date with the Github API.

File: templates/cli/scoop/appwrite.json.twig
Comment (CEbbinghaus) [inline templates/cli/scoop/appwrite.json.twig:12]: This should be changed to include ARM now that Scoop supports it ScoopInstaller/Scoop#3146.

File: templates/cli/README.md.twig
Comment (CEbbinghaus) [inline templates/cli/README.md.twig:66]: Gotta remove the `/#/`since its not necessary

**Review by CEbbinghaus — COMMENTED** [review]
Not quite ready to be merged. Will try and get these changes in today to bring it up to date. Also of note is that this will generate the json file every single version but only 1 will ever be committed to the main bucket scoop maintains as it self updates the json through the `"checkver"` property.

**Comment by lohanidamodar** [comment]
@CEbbinghaus thank you for your immediate response. Please make the necessary changes and request another review.

**Comment by CEbbinghaus** [comment]
@lohanidamodar who should I request the review from?

**Review by stnguyen90 — COMMENTED** [review]
One suggestion.

**Review by stnguyen90 — APPROVED** [review]

**Comment by CEbbinghaus** [comment]
So this is ready to merge? The failed Travis build seems to be more of an amnesty rather than due to my changes. Is anything hold it up?

**Review by stnguyen90 — CHANGES_REQUESTED** [review]
> So this is ready to merge? The failed Travis build seems to be more of an amnesty rather than due to my changes. Is anything hold it up?

@CEbbinghaus would you please update this branch with the latest from master? There may be something on master that will fix the CI.

**Comment by CEbbinghaus** [comment]
Updated with the latest from master. Lets hope it fixes the CI task :3

**Review by stnguyen90 — APPROVED** [review]

**Pull request merged**
